### PR TITLE
Implement day_name and month_name to match pandas

### LINKED ISSUE #12407: [FEA] DatetimeProperties day_name matching pandas
**Is your feature request related to a problem? Please describe.**
converting code with `import cudf as pd`

```
>>> import cudf as pd
>>> pd.__version__
'22.12.01'
>>> df = pd.Series(pd.date_range("1984", freq='s', periods=6))
>>> df.to_pandas().dt.day_name()
0    Sunday
1    Sunday
2    Sunday
3    Sunday
4    Sunday
5    Sunday
dtype: object
>>> df.dt.day_name()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'DatetimeProperties' object has no attribute 'day_name'
>>> type(df.dt)
<class 'cudf.core.series.DatetimeProperties'>
```

**Describe the solution you'd like**
implementation of `day_name()` matching https://pandas.pydata.org/docs/reference/api/pandas.Series.dt.day_name.html

### PR DESCRIPTION
**Pull request by btepera**
## Description
This PR implements the `month_name` and `day_name` datetime methods, matching the equivalent [month_name](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.month_name.html) and [day_name](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.day_name.html) methods from pandas.

Currently this is implemented just for English locale, though it could be expanded to include additional languages in the future. 

Closes #12407 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/index.py [hunk 1]
```diff
@@ -2078,6 +2078,54 @@ def quarter(self):
         res = extract_quarter(self._values)
         return Index(res, dtype="int8")
 
+    @_cudf_nvtx_annotate
+    def day_name(self, locale=None):
```
Comment (mroeschke) [inline python/cudf/cudf/core/index.py:2082]: Could you ensure a non `None` locale will raise a `NotImplementedError`?

File: python/cudf/cudf/core/index.py [hunk 2]
```diff
@@ -2078,6 +2078,54 @@ def quarter(self):
...
+        """
+        Return the day names. Currently supports English locale only.
+
+        Examples
+        --------
+        >>> import pandas as pd
+        >>> import cudf
+        >>> datetime_index = cudf.Index(pd.date_range("2016-12-31",
```
Comment (mroeschke) [inline python/cudf/cudf/core/index.py:2090]: Could you just use `cudf.date_range` here?

File: python/cudf/cudf/tests/test_datetime.py [hunk 3]
```diff
@@ -2373,3 +2373,69 @@ def test_date_range_tz():
...
+    # Series
+    ps = pd.Series(data, dtype="datetime64[s]")
+    gs = cudf.from_pandas(ps)
+
+    expect = ps.dt.day_name()
+    got = gs.dt.day_name()
+
+    assert_eq(expect, got)
+
+    # DatetimeIndex
```
Comment (mroeschke) [inline python/cudf/cudf/tests/test_datetime.py:2401]: Could you make the `DatetimeIndex` a separate test case?

File: python/cudf/cudf/core/index.py [hunk 4]
```diff
@@ -2078,6 +2078,54 @@ def quarter(self):
...
+        DatetimeIndex(['2016-12-31', '2017-01-01', '2017-01-02', '2017-01-03',
+                    '2017-01-04', '2017-01-05', '2017-01-06', '2017-01-07',
+                    '2017-01-08'],
+                    dtype='datetime64[ns]')
+        >>> datetime_index.day_name()
+        Index(['Saturday', 'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday',
+               'Friday', 'Saturday'], dtype='object')
+        """
+        days = self._get_dt_field("weekday")
+        day_names = cudf.core.tools.datetimes._get_date_name_field(
```
Comment (mroeschke) [inline python/cudf/cudf/core/index.py:2102]: I think it might be better to cast the column to string and use the column's `find_and replace`

**Comment by btepera** [comment]
Thanks for the review @mroeschke ! Pushed a few changes to address your feedback.

**Comment by mroeschke** [comment]
Thanks for your initial work here @btepera

**Pull request merged**